tracer/src/Datadog.Trace/Ci/CIEnvironmentValues.cs

```diff
@@ -382,8 +410,64 @@ internal void ReloadEnvironmentData()
             // **********
             Branch = GetEnvironmentVariableIfIsNotEmpty("DD_GIT_BRANCH", Branch);
             Tag = GetEnvironmentVariableIfIsNotEmpty("DD_GIT_TAG", Tag);
-            Repository = GetEnvironmentVariableIfIsNotEmpty("DD_GIT_REPOSITORY_URL", Repository);
-            Commit = GetEnvironmentVariableIfIsNotEmpty("DD_GIT_COMMIT_SHA", Commit);
+            Repository = GetEnvironmentVariableIfIsNotEmpty(
+                "DD_GIT_REPOSITORY_URL",
+                Repository,
+                (value, defaultValue) =>
+                {
+                    if (value is not null)
+                    {
+                        value = value.Trim();
+                        if (value.Length == 0)
+                        {
+                            Log.Error("DD_GIT_REPOSITORY_URL is set with an empty value, defaulting to '{default}'", defaultValue);
```

Comment:
nit: If `DD_GIT_REPOSITORY_URL` is empty or invalid, but the repo couldn't be extracted, you'll get two errors in both cases. Might be nice to combine these? e.g something like

```csharp
if(string.IsNullOrEmpty(defaultValue))
{
    Log.Error("DD_GIT_REPOSITORY_URL is set with an empty value, and the Git repository could not be automatically extracted");
}
else
{
    Log.Error("DD_GIT_REPOSITORY_URL is set with an empty value, defaulting to '{default}'", defaultValue);
}

return false;
```

Reply:
done in [5f6129b](https://github.com/DataDog/dd-trace-dotnet/pull/3560/commits/5f6129b79c10083f07e4d9454611bf42eb428b22)